sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlAsyncClient.java

```diff
@@ -185,32 +204,311 @@ HttpPipeline getHttpPipeline() {
+     *
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} or {@link String roleDefinitionName}
+     * in the {@link SetKeyVaultRoleDefinitionOptions options} object are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<KeyVaultRoleDefinition> setRoleDefinition(SetKeyVaultRoleDefinitionOptions options) {
+        return setRoleDefinitionWithResponse(options).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Creates or updates a {@link KeyVaultRoleDefinition}.
+     *
+     * @param options Object representing the configurable options to create or update a
+     * {@link KeyVaultRoleDefinition role definition}.
+     *
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the
+     * created or updated {@link KeyVaultRoleDefinition}.
+     *
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} or {@link String roleDefinitionName}
+     * in the {@link SetKeyVaultRoleDefinitionOptions options} object are {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<KeyVaultRoleDefinition>> setRoleDefinitionWithResponse(
+        SetKeyVaultRoleDefinitionOptions options) {
+
+        return withContext(context -> setRoleDefinitionWithResponse(options, context));
+    }
+
+    /**
+     * Creates or updates a {@link KeyVaultRoleDefinition}.
+     *
+     * @param options Object representing the configurable options to create or update a
+     * {@link KeyVaultRoleDefinition role definition}.
+     * @param context Additional context that is passed through the HTTP pipeline during the service call.
+     *
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains the
+     * created or updated {@link KeyVaultRoleDefinition}.
+     *
+     * @throws NullPointerException If the {@link KeyVaultRoleScope role scope} or {@link String roleDefinitionName}
+     * in the {@link SetKeyVaultRoleDefinitionOptions options} object are {@code null}.
+     */
+    Mono<Response<KeyVaultRoleDefinition>> setRoleDefinitionWithResponse(SetKeyVaultRoleDefinitionOptions options,
+                                                                         Context context) {
+        Objects.requireNonNull(options,
+            String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
+                "'options'"));
+        Objects.requireNonNull(options.getRoleScope(),
+            String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
+                "'options.getRoleScope()'"));
+        Objects.requireNonNull(options.getRoleDefinitionName(),
+            String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
+                "'options.getRoleDefinitionName()'"));
```

Comment:
Return `Mono.error()` instead of throwing exceptions when the method returns a reactive publisher.

Reply:
This private method is called by `setRoleDefinitionWithResponse` only, which in turn uses `withContext`. Would this not be handled by the `onError` part of the publisher?

Reply:
Now all these null checks are wrapped using `Mono.error()`.